### PR TITLE
[ARM] Fix bf32 and tf32 precision for tensordot unit test

### PR DESCRIPTION
**Pull request by robert-hardwick**
Fixes unit test failure on aarch64 ( neoverse-v1 )

cc @malfet @snadampal @milpuz01

### CODE REVIEW

**Review by malfet — CHANGES_REQUESTED** [review]
Sorry, but this reduces the precision of the test not just on ARM platform, but across the board.  And it would be nice to add some explanation on why it is needed on Linux arm, but not on say Mac arm? Is this because of the bug/accuracy problem in the ACL library? If so, it would be nice to reference an issue/commit message where this decision was made

**Comment by robert-hardwick** [comment]
Hi, of course, yeah i completely forgot to add the context here. We see the following error on neoverse-v1 ( c7g ).

```
AssertionError: Tensor-likes are not close!
Mismatched elements: 158 / 252 (62.7%)
Greatest absolute difference: 0.03098297119140625 at index (0, 1, 1, 4) (up to 0.005 allowed)
Greatest relative difference: 0.2863367199897766 at index (1, 1, 1, 4) (up to 1.3e-06 allowed)
To execute this test, run the following from the base repo dir:
    python test/test_linalg.py TestLinalgCPU.test_tensordot_cpu
This message can be suppressed by setting PYTORCH_PRINT_REPRO_ON_FAILURE=0
```

which is this assertion here https://github.com/pytorch/pytorch/blob/75fbcc57439339d27841b4b81ff6a31537a1bac5/test/test_linalg.py#L8444

A single calculation in bf32 has the potential to lose up to 0.0078125 in precision ( although in practice unlikely ) 

```
>>> torch.finfo(torch.bfloat16).eps
0.0078125
```

but since each of the values in the matrix performs a 4 * 5 reduction, i believe the maximum possible error is 0.0078125 * 4 * 5 but in practice this was consistently < ( 0.0078125 * 4 * 5 ) / 2. Hence the choice of 0.8

Admittedly i should have left tf32 alone, I can amend the patch.

TLDR; bf32 tolerances for summation calculations is highly dependend on the number of summations. E.g. in a matmul the bf32 tolerance is dependent on the inner matrix dimension.

**Comment by robert-hardwick** [comment]
Providing an update here.... I've been doing some statistical analysis of the abs diff. I have run the failing test 200 times to collect a distribution of the Max Absolute Difference and here is the result. It seems to stabalize at 0.069. Even with 1000 runs, the most abs diff we get is 0.069.  I need to check the math to confirm it makes sense.

plot_max_abs

However, I tried the same with Intel ( I can see the amx_bf16 flag is enabled ) and i can see that [the bf16 attr](attr-scratchpad:user attr-fpmath:bf16) is mentioned in ONEDNN, but it shows 0 difference compared to the reference implementation.

```
onednn_verbose,primitive,create:cache_hit,cpu,matmul,brg_matmul:avx512_core,undef,src_f32::blocked:ab::f0 wei_f32::blocked:ab::f0 dst_f32::blocked:ab::f0,attr-scratchpad:user attr-fpmath:bf16,,6x20:20x42,0.0012207
onednn_verbose,primitive,exec,cpu,matmul,brg_matmul:avx512_core,undef,src_f32::blocked:ab::f0 wei_f32::blocked:ab::f0 dst_f32::blocked:ab::f0,attr-scratchpad:user attr-fpmath:bf16,,6x20:20x42,0.00195312
max_abs_diff 0.0
max_rel_diff 0.0
```

So... either the reference implementation is also doing bf16->fp32 implicit conversion on intel, or attr-fpmath:bf16 isn't having an effect on intel.

**Comment by robert-hardwick** [comment]
Is it possible that np.tensordot makes use of bf16 on intel?

**Review by malfet — APPROVED** [review]
